src/event-stream/tokens-contract-handler.ts

```diff
@@ -0,0 +1,417 @@
+      const contractCallSymbol = await this.makeReadOnlyContractCall('get-symbol', []);
+      const contractCallDecimals = await this.makeReadOnlyContractCall('get-decimals', []);
+
+      //check parse clarity values
+      const nameCV = this.checkAndParseString(contractCallName);
+      const uriCV = this.checkAndParseOptionalString(contractCallUri);
+      const symbolCV = this.checkAndParseString(contractCallSymbol);
+      const decimalsCV = this.checkAndParseUintCV(contractCallDecimals);
+
+      let metadata: FTTokenMetadata = { name: '', description: '', image: '' };
+      //fetch metadata from the uri if possible
+      if (uriCV) metadata = await this.getMetadataFromUri<FTTokenMetadata>(uriCV.data);
+      const { name, description, image } = metadata;
+      const fungibleTokenMetadata: DbFungibleTokenMetadata = {
+        name: nameCV ? nameCV.data : name, //prefer the on-chain name
+        description: description,
+        image_uri: image,
+        image_canonical_uri: uriCV ? uriCV.data : '',
+        symbol: symbolCV ? symbolCV.data : '',
+        decimals: decimalsCV ? Number(decimalsCV.value) : 0,
+        contract_id: `${this.contractAddress}.${this.contractName}`,
+      };
+
+      //store metadata in db
+      await this.storeFTMetadata(fungibleTokenMetadata);
+    } catch (error) {
+      logger.error('erro: error handling FT contract', error);
+    }
+  }
+
+  /**
+   * fetch Non Fungible contract metadata
+   */
+  private async handleNFTContract() {
+    try {
+      const contractCallTokenId = await this.makeReadOnlyContractCall('get-last-token-id', []);
+      const tokenId = this.checkAndParseUintCV(contractCallTokenId);
+
+      if (tokenId) {
+        const contractCallUri = await this.makeReadOnlyContractCall('get-token-uri', [tokenId]);
+        const uriCV = this.checkAndParseOptionalString(contractCallUri);
+
+        let metadata: NFTTokenMetadata = { name: '', description: '', imageUrl: '' };
+        if (uriCV) metadata = await this.getMetadataFromUri<NFTTokenMetadata>(uriCV.data);
+
+        const nonFungibleTokenMetadata: DbNonFungibleTokenMetadata = {
+          name: metadata.name,
+          description: metadata.description,
+          image_uri: metadata.imageUrl,
+          image_canonical_uri: uriCV ? uriCV.data : '',
+          contract_id: `${this.contractAddress}.${this.contractName}`,
+        };
+
+        await this.storeNFTMetadata(nonFungibleTokenMetadata);
+      }
+    } catch (error) {
+      logger.error('erro: error handling NFT contract', error);
+    }
+  }
+
+  /**helpng method for creating http url */
+  private makeHostedUrl(uri: string): string {
+    if (uri.includes('http')) return uri;
+    const parsedURI = URI.parse(uri);
+    return `${PUBLIC_IPFS}/${parsedURI.host}${parsedURI.path}`;
+  }
+
+  async makeApiCall<Type>(url: string): Promise<Type> {
+    const result = await fetch(url);
```

Comment:
Is the stacks-node guaranteed to respond to this `/v2/...` request while its event dispatcher is waiting on the API for a response to the `/new_burn_block` event request, which this path is currently holding up? cc @kantai

Reply:
No -- the relayer and p2p thread both stalled while the event dispatcher is transmitting data to the event observer.